### PR TITLE
Update pattern filenames, and titles

### LINKED ISSUE #244: Audit patterns filenames
**Description**

Our goal is to make the naming as intuitive as possible. To achieve this, we need to audit the patterns' names and define proper naming for the different templates in the `patterns` folder. We can check how it was done for the past default themes with patterns so that users are familiar with how we name things out of the box.

I believe we should consider the convention that was historically established in WordPress themes, but as prefixes: `single-*`, `page-*`, `archive-*`, `home-*`, `search-*`. Thinking also of `template-*` as an option.

We also need to find some convention for the other isolated patterns the theme is packed with, an idea could be thinking of the main category as the prefix.

### PR DESCRIPTION
**Pull request by jasmussen**
**Description**

Starts work at, maybe fixes #244. This PR does three things:

1. It renames pattern filenames according to [the proposed format here](https://github.com/WordPress/twentytwentyfive/issues/244#issuecomment-2377605175).
2. It updates pattern _slugs_ to reference match the filenames, and updates any includes to point to the new slugs.
3. It updates titles to use numbers instead of text (3 columns instead of three columns), sentence case instead of titlecase, and comma instead of a colon for prefixes.

This needs a good bit of testing to see that patterns work and notably item 2, to make sure I've captured every template that includes other patterns with the correct slug.

Feel free to push changes to this PR.

### CODE REVIEW

File: patterns/page-business-home.php [hunk 1]
```diff
@@ -18,9 +18,9 @@
...
 <div class="wp-block-group alignfull">
 	<!-- wp:pattern {"slug":"twentytwentyfive/cta-centered-heading"} /-->
-	<!-- wp:pattern {"slug":"twentytwentyfive/overlap-images-and-paragraph"} /-->
-	<!-- wp:pattern {"slug":"twentytwentyfive/services-three-columns"} /-->
-	<!-- wp:pattern {"slug":"twentytwentyfive/testimonials-review-large-image-right"} /-->
+	<!-- wp:pattern {"slug":"twentytwentyfive/overlapped-images"} /-->
+	<!-- wp:pattern {"slug":"twentytwentyfive/services-3-col"} /-->
+	<!-- wp:pattern {"slug":"twentytwentyfive/testimonials-large"} /-->
 	<!-- wp:pattern {"slug":"twentytwentyfive/pricing-2-col"} /-->
 	<!-- wp:pattern {"slug":"twentytwentyfive/newsletter-sign-up"} /-->
```
Comment (juanfra) [inline patterns/page-business-home.php]:
```suggestion
	<!-- wp:pattern {"slug":"twentytwentyfive/cta-newsletter"} /-->
```

File: patterns/page-landing-book.php [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?php
 /**
- * Title: Landing page for Book
+ * Title: Landing page for book
  * Slug: twentytwentyfive/page-landing-book
  * Categories: twentytwentyfive_page, featured
  * Keywords: starter
```
Comment (juanfra) [inline patterns/page-landing-book.php:6]: In this one we also need to update the `newsletter-signup` to `cta-newsletter`

File: patterns/template-search-news-blog.php [hunk 3]
```diff
@@ -30,7 +30,7 @@
 	<!-- /wp:group -->
 	<!-- wp:group {"align":"wide","layout":{"type":"default"}} -->
 	<div class="wp-block-group alignwide">
-		<!-- wp:pattern {"slug":"twentytwentyfive/news-blog-query-loop"} /-->
```
Comment (juanfra) [inline patterns/template-search-news-blog.php:33]: This template also needs the renaming of the slug.

**Review by juanfra — COMMENTED** [review]
Thanks, Joen! Works well for me. I found a few tiny things I'm going to push changes for.

File: patterns/media-instagram-grid.php [hunk 4]
```diff
@@ -1,8 +1,8 @@
 <?php
 /**
  * Title: Instagram grid
- * Slug: twentytwentyfive/instagram-grid
- * Categories: media, featured
+ * Slug: twentytwentyfive/media-instagram-grid
+ * Categories: media, gallery, featured
```
Comment (carolinan) [inline patterns/media-instagram-grid.php:5]: ⭐

File: patterns/services-3-col.php [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <?php
 /**
- * Title: Services - Three Columns
- * Slug: twentytwentyfive/services-three-columns
+ * Title: 3 columns
+ * Slug: twentytwentyfive/services-3-col
```
Comment (carolinan) [inline patterns/services-3-col.php:4]: Is this meant to be Services, 3 columns, like the Pricing, 3 columns?

File: patterns/template-archive-vertical-header-blog.php [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <?php
 /**
  * Title: Archive for the right aligned blog
```
Comment (carolinan) [inline patterns/template-archive-vertical-header-blog.php]: I think all the right-aligned template names could use some adjusting.
Is it right-aligned, right aligned, or, is it the vertical header that should be in focus?

File: patterns/services-3-col.php (hunk 5 above)
Comment (jasmussen) [reply patterns/services-3-col.php:4]: Yes. One moment.

File: patterns/template-archive-vertical-header-blog.php (hunk 6 above)
Comment (jasmussen) [reply patterns/template-archive-vertical-header-blog.php]: I'll take a stab.

**Comment by jasmussen** [comment]
Pushed some "aligned" improvements, and added "Services, " as a prefix to the 3 columns.

**Pull request merged**